### PR TITLE
Service for data JSON files

### PR DESCRIPTION
**Pull request by donna-marijne**
Added methods to AppConfiguration to return the contents of the data/*.json files, and update them. This is used in the new documentation site to inject custom site layout.

https://jira.autonomy.com/browse/EL-2759

### CODE REVIEW

File: docs/app/components/navigation-bar-search/navigation-bar-search.component.ts [hunk 1]
```diff
@@ -34,46 +35,35 @@ export class NavigationBarSearchComponent {
...
-    constructor(private router: Router, private navigation: NavigationService, private versionService: VersionService) {
-
-        // get version
-        this.versionService.version.subscribe((value: Version) => {
-            this.data = [];
-            this.data = this.data.concat(this.getSearchResults(componentsData));
-            this.data = this.data.concat(this.getSearchResults(cssData));
-            this.data = this.data.concat(this.getSearchResults(chartsData));
-        });
+    constructor(private router: Router, private navigation: NavigationService, private versionService: VersionService, private _appConfig: AppConfiguration) {
```
Comment (chris-brennan) [inline docs/app/components/navigation-bar-search/navigation-bar-search.component.ts]: Should private variables be prefixed with an underscore. i.e. `private _router: Router` ?

File: docs/app/pages/customize/customize.component.ts [hunk 2]
```diff
@@ -11,13 +12,13 @@ export class CustomizePageComponent {
     sections: ICustomisePageSection[];
     variables: any;
 
-    constructor(private lessService: LessService) {
+    constructor(private lessService: LessService, private _appConfig: AppConfiguration) {
```
Comment (chris-brennan) [inline docs/app/pages/customize/customize.component.ts]: underscore prefix?

File: docs/app/services/app-configuration/app-configuration.service.ts [hunk 3]
```diff
@@ -3,19 +3,39 @@ import { Injectable } from '@angular/core';
...
     }
 
     private getConfig() {
         switch (process.env.ENV) {
             case 'development':
-                return require('../../data/config.dev.json');
+                return this._data['config.dev'];
         }
-        return require('../../data/config.json');
+        return this._data['config'];
```
Comment (chris-brennan) [inline docs/app/services/app-configuration/app-configuration.service.ts]: I'm not sure but could this use dot notation? Or is bracket notation preferred here?

File: docs/app/components/navigation-bar-search/navigation-bar-search.component.ts (hunk 1 above)
Comment (donna-marijne) [reply docs/app/components/navigation-bar-search/navigation-bar-search.component.ts]: Yes. This is what I used. `private _appConfig: AppConfiguration`

File: docs/app/services/app-configuration/app-configuration.service.ts (hunk 3 above)
Comment (donna-marijne) [reply docs/app/services/app-configuration/app-configuration.service.ts]: It's a key-value table, so dot is not preferred. Most keys include a period or hyphen so using property syntax would be impossible.

Comment (chris-brennan) [reply docs/app/services/app-configuration/app-configuration.service.ts]: OK I didn't know that, thanks.

**Comment by chris-brennan** [comment]
looks good to me

**Pull request merged**